composer/algorithms/augmix/augmix.py

```diff
@@ -251,15 +251,19 @@ def __init__(self,
         self._transformed_datasets = weakref.WeakSet()
 
     def match(self, event: Event, state: State) -> bool:
-        return event == Event.FIT_START and state.train_dataloader.dataset not in self._transformed_datasets
+        if event != Event.FIT_START:
+            return False
+        assert state.dataloader is not None, "dataloader should be defined on fit start"
+        return state.dataloader.dataset not in self._transformed_datasets
 
     def apply(self, event: Event, state: State, logger: Logger) -> None:
         am = AugmentAndMixTransform(severity=self.severity,
                                     depth=self.depth,
                                     width=self.width,
                                     alpha=self.alpha,
                                     augmentation_set=self.augmentation_set)
-        dataset = state.train_dataloader.dataset
+        assert state.dataloader is not None, "dataloader should be defined on fit start"
```

Comment:
I added the getter/setter on state so `state.dataloader_len` can be automatically whenever the dataloader is swapped. However, I would still prefer to require the caller to handle a nullable dataloader, rather than raising an exception (such as a  RuntimeError) if it is not set, as that will require the caller code to be more robust.